### PR TITLE
JDK-8282777: Create a Regression test for JDK-4515031

### PR DESCRIPTION
**Pull request by srmandal**
Create a Regression test for [JDK-4515031](https://bugs.openjdk.java.net/browse/JDK-4515031)

The issue indicates the need for a a getAccessibleDescription being implemented for the JFileChooser.
The test added verifies the same.
This review is for migrating tests from a closed test suite to open.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8282777](https://bugs.openjdk.org/browse/JDK-8282777): Create a Regression test for JDK-4515031

### Reviewers
 * [Sergey Bylokhov](https://openjdk.org/census#serb) (@mrserb - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/7738/head:pull/7738` \
`$ git checkout pull/7738`

Update a local copy of the PR: \
`$ git checkout pull/7738` \
`$ git pull https://git.openjdk.org/jdk pull/7738/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7738`

View PR using the GUI difftool: \
`$ git pr show -t 7738`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/7738.diff">https://git.openjdk.org/jdk/pull/7738.diff</a>

</details>

### CODE REVIEW

File: test/jdk/javax/accessibility/4515031/JFileChooserAccessibleDescriptionTest.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    }
+
+    public static void doTest() throws Exception {
+        try {
+            SwingUtilities.invokeAndWait(() -> createGUI());
+            robot = new Robot();
+            robot.setAutoDelay(200);
+            robot.setAutoWaitForIdle(true);
+
+            robot.mouseMove(jButton.getLocationOnScreen().x + jButton.getSize().width / 2,
```
Comment (mrserb) [inline test/jdk/javax/accessibility/4515031/JFileChooserAccessibleDescriptionTest.java:72]: The jButton is a Swing component, it should be accessed on EDT.
Please split the long lines, 80<= chars.

**Comment by srmandal** [comment]
@mrserb can you please review the changes after incorporating the earlier feedback ?

**Comment by mrserb** [comment]
Please confirm that mach5 is green after this fix.

**Comment by srmandal** [comment]
Testing:
The tests ran successfully on Mach5 with multiple runs (30) on windows-x64, linux-x64 and macos-x64.

**Comment by srmandal** [comment]
Yes. The test ran successfully on Mach5 with multiple runs (30) on windows-x64, linux-x64 and macos-x64.

File: test/jdk/javax/accessibility/JFileChooserAccessibleDescriptionTest.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+            robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+
+            SwingUtilities.invokeAndWait(() -> description =
+                jFileChooser.getAccessibleContext().getAccessibleDescription());
+
+            robot.keyPress(KeyEvent.VK_ALT);
+            robot.keyPress(KeyEvent.VK_C);
+            robot.keyRelease(KeyEvent.VK_C);
+            robot.keyRelease(KeyEvent.VK_ALT);
```
Comment (azuev-java) [inline test/jdk/javax/accessibility/JFileChooserAccessibleDescriptionTest.java:89]: What is the purpose of the pressing Alt-C here? Looks like on Mac OS X it does nothing and when i comment them out the test runs in the same way as with it.

Comment (srmandal) [reply test/jdk/javax/accessibility/JFileChooserAccessibleDescriptionTest.java]: As suggested, I have removed this redundant code and verified it on the windows,mac and linux platforms to be working fine.

**Comment by srmandal** [comment]
Hi! Can someone please take a look at this PR and provide their valuable feedback ?

**Review by mrserb — APPROVED** [review]

**Pull request closed without merging**